www/ebooks/index.php

```diff
@@ -35,12 +35,14 @@
 		$tags = [];
 	}
 
-	$ebooks = Library::FilterEbooks($query != '' ? $query : null, $tags, $sort);
+	$result = Library::FilterEbooks($query != '' ? $query : null, $tags, $sort, $page, $perPage);
+	/** @var array<Ebook> $ebooks */
```

Comment:
Once you use array shapes in the phpdoc for `FilterEbooks`, you should be able to remove these inline docs because PHPStan will already know the return value.

Reply:
Yep, done.